pywbem_mock/_dmtf_schema.py

```diff
@@ -0,0 +1,392 @@
+            ValueError if the schema cannon be retrieved from the DMTF web site.
+            ValueError or if the 'schema_dir' differs from the version of the
+            schema currently set into the 'schema_dir'
+
+        """
+        if not isinstance(schema_ver, tuple):
+            raise ValueError('schema_ver must be tuple not %s' % schema_ver)
+        if len(schema_ver) != 3:
+            raise ValueError('DMTF Schema must have 3 integer components, '
+                             '(major, minor, rev) not %s' % schema_ver)
+        schema_type = 'Experimental' if use_experimental else 'Final'
+        self._schema_ver = schema_ver
+        self._schema_dir = schema_dir
+
+        mof_dir = 'mof%s%s' % (schema_type, self.schema_ver_str)
+        self._schema_mof_dir = os.path.join(self._schema_dir, mof_dir)
+
+        self._dmtf_schema_ver = 'cim_schema_%s' % (self.schema_ver_str)
+        self._mof_zip_bn = '%s%s-MOFs.zip' % (self._dmtf_schema_ver,
+                                              schema_type)
+        self._mof_zip_url = \
+            'http://www.dmtf.org/standards/cim/cim_schema_v%s%s%s/%s' % \
+            (schema_ver[0], schema_ver[1], schema_ver[2], self._mof_zip_bn)
+        self._schema_mof_bld_name = self._dmtf_schema_ver + '.mof'
+
+        self._mof_zip_filename = os.path.join(self.schema_dir, self._mof_zip_bn)
+        self._schema_mof_filename = os.path.join(self.schema_mof_dir,
+                                                 self._schema_mof_bld_name)
+
+        self.verbose = verbose
+
+        # setup the schema
+        self._setup_dmtf_schema()
+
+    @property
+    def schema_ver(self):
+        """
+        :func:`py:tuple` of 3 integers with major version, minor version, and
+        revision of the DMTF Schema installed
+        """
+        return self._schema_ver
+
+    @property
+    def schema_ver_str(self):
+        """
+        :term:`string` with the DMTF schema version in the form
+        <major version>.<minor version>.<revison>.
+        """
+        return '%s.%s.%s' % (self.schema_ver[0], self.schema_ver[1],
+                             self.schema_ver[2])
+
+    @property
+    def schema_dir(self):
+        """
+        :term:`string` defining the directory in which the DMTF CIM Schema
+        mof is downloaded and expanded.
+        """
+        return self._schema_dir
+
+    @property
+    def schema_mof_dir(self):
+        """
+        :term:`string` defining the directory in which the DMTF CIM Schema
+        mof is defined. This includes the expansion of the individual schema
+        mof files. This property is very useful since it can be used as
+        the search path for compiling classes in the DMTF schema.
+        """
+        return self._schema_mof_dir
+
+    @property
+    def schema_mof_filename(self):
+        """
+        :term:`string` defining the the path for the DMTF schema
+        mof top level file name which includes the pragmas that define
```

Comment:
On "DMTF schema mof top level file name": Can we expand that a little? Also, it is not the path of the file name, but the path of the file. How about: "The file path of the top level MOF file of the DMTF schema".